### PR TITLE
Implemented variables that allow customizing the format for the window title

### PR DESCRIPTION
**Pull request by BrainStone**
This addresses #2.

In short this adds the two variables `ZSH_WINDOW_TITLE_IDLE` and `ZSH_WINDOW_TITLE_ACTIVE`, which contain the formats for when no program is running and when a program is running respectively.  
Per default they contain the formats that were used previously.

This also includes updated documentation.

The change has been tested and seems to work fine.

Note I also added a .gitignore, that roughly suited my development environment. If you don't want it, I'll be happy to remove it.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -69,7 +69,7 @@ By default macOS's Terminal app will control the window title. Disable that by:
 By default OMZ will control the terminal window title. Disable that by adding
 
 ```shell
-DISABLE_AUTO_TITLE=”true”
+DISABLE_AUTO_TITLE="true"
```
Comment (olets) [inline README.md:72]: ooh good catch

**Review by olets — CHANGES_REQUESTED** [review]
Nice, it's cool you got this working!

I'm not ready to merge as is, because for now I want to keep the plugin more opinionated and beginner-proof. From https://github.com/olets/zsh-window-title/issues/2#issuecomment-1014054568 looks like yours is different enough —and you're expert enough— to roll your own to meet your exact needs!

```shell
window-title:precmd() {
  echo -ne "\033]0;$(print -P …)\007"
}
window-title:preexec() {
  echo -ne "\033]0;$(print -P …)\007"
}
autoload -U add-zsh-hook
add-zsh-hook precmd window-title:precmd
add-zsh-hook preexec window-title:preexec
```

If you're interested in reworking this PR, I do still like the idea of supporting a single custom prefix to put in front of both the active and idle titles.

If you do, yeah let's remove the gitignore since those files (with the possible exception of the *.zwc) are about your environment not something this repo would have. A lot of that might be better suited to your global gitignore (if that's new to you, check out [the article one of my coworkers wrote](https://www.viget.com/articles/global-gitignore-files-are-cool-and-so-are-you-1/)).

**Comment by BrainStone** [comment]
Fair enough.

Would you be ok with an approach that adds both sets of variables?  
My thinking being that once the features discussed in #1 are implemented people that want to fully customize their window title can still benefit from the threshold (mechanic)?  
I would put the two big variables in their own section advising against changing them, unless people know what they are doing.

And funnily enough I did not know about a global .gitignore.

**Comment by BrainStone** [comment]
Alright. There we go!

What do you think of it like this?

**Comment by olets** [comment]
This is cool but I still think opening the title itself for customization is too powerful for zsh-window-title. My recommendation for anyone who finds they want to get really custom is to ditch a plugin and write their own little snippet like the one above — it's a good intro to shell scripting (customizing my window title was one of the first things I ever wrote for my zshrc…… and in the years since I've spent countless full days in shellscriptland. So… actually maybe my top recommendation should be be careful of opening that door 😆) and means complete flexibility.

Thanks again for the pull request @BrainStone! I'm honored that you gave zsh-window-title that development time. And like I said before I'd still consider a prefix or suffix PR.

**Pull request closed without merging**